File: 백준/Silver/10828. 스택/스택.cc
#include<bits/stdc++.h>
using namespace std;

int main(){
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int N;
    cin>>N;

    stack<int> S;
    while(N--){
        string str;
        cin>>str;

        if(str=="push"){
            int a;
            cin>>a;

            S.push(a);
        }
        else if(str=="pop"){
            if(S.empty()){
                cout<<-1<<'\n';
            }
            else{
                cout<<S.top()<<'\n';
                S.pop();
            }
        }
        else if(str=="size"){
            cout<<S.size()<<'\n';
        }
        else if(str=="empty"){
            cout<<(int)S.empty()<<'\n';
        }
        else{
            if(S.empty()){
                cout<<-1<<'\n';
            }
            else{
                cout<<S.top()<<'\n';
            }
        }
    }
}

File: 백준/Silver/10828. 스택/스택.java
import java.util.*;
import java.io.*;

public class Main {
    static Stack<Integer> stack = new Stack<>();
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer stk = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(stk.nextToken());

        for (int i = 0; i < n; i++) {
            stk = new StringTokenizer(br.readLine(), " ");
            String str = stk.nextToken();
            if (str.equals("push")) {
                int tmp = Integer.parseInt(stk.nextToken());
                push(tmp);
            } else if (str.equals("pop")) {
                pop();
            } else if (str.equals("size")) {
                size();
            } else if (str.equals("empty")) {
                empty();
            } else {
                top();
            }
        }
    }

    static void push(int x) {
        stack.push(x);
    }

    static void pop() {
        if (stack.isEmpty()) {
            System.out.println(-1);
        } else {
            System.out.println(stack.peek());
            stack.pop();
        }
    }

    static void size() {
        System.out.println(stack.size());
    }

    static void empty() {
        if (stack.isEmpty()) {
            System.out.println(1);
        } else {
            System.out.println(0);
        }
    }

    static void top() {
        if (stack.isEmpty()) {
            System.out.println(-1);
        } else {
            System.out.println(stack.peek());
        }
    }
}